### DISCUSSION TITLE
Batch Push Interval is not working as expected

### DISCUSSION THREAD

**Discussion by RamakrishnanRajaram** [aws/aws-logging-dotnet#236] [Q&A] [answered]
We are pushing the logs from dotnet API to the CloudWatch via AWS Lambda. We wanted to send the logs at frequent intervals to keep the lambda warm. So we decided to reduce the time for Batch Push Interval, which is defaulted to 3 seconds. Though we set the batch push interval as 100 ms, still the logs are pushed only at the interval of 3 seconds.

var config = builder.Configuration.GetAWSLoggingConfigSection();
config.Config.BatchPushInterval = TimeSpan.FromMilliseconds(100.0);

**Comment by normj** [comment, marked as answer]
Decreasing the `BatchPushInterval` would have no effect on keeping the Lambda function warm. The logs are pushed in a background thread. The function will go "cold" as soon as the thread that invoke the function handler returns.
